File: mainframe_migration/examples/generated_targets/python-sqlite/jobs/JOB002.py
"""JOB002 — migrated source job.
Run every JCL step in order and preserve the original business behavior.
Inputs/outputs are the source DD bindings; database access is selected separately.
Source: jcl/JOB002.jcl
"""
from decimal import Decimal, localcontext
from _runtime import Fields, relation

def step_S010(ctx):
    """Execute S010 / SORT with the original DD bindings."""
    dd = {'SORTIN': {'line': 3, 'dsn': 'SAMPLE.ACCEPT', 'disp': 'SHR'}, 'SORTOUT': {'line': 4, 'dsn': 'SAMPLE.SORTED', 'disp': '(NEW,CATLG,DELETE)'}, 'SYSIN': {'line': 5, 'inline': ' SORT FIELDS=(1,6,CH,A)\n OPTION EQUALS'}}
    ctx.sort(dd["SORTIN"], dd["SORTOUT"], [[1, 6, 'CH', 'A']], ctx.payload.get("max_sort_records", 100000))
    ctx.end_step()

def run(ctx):
    """Run all source steps; errors stop the job instead of skipping work."""
    with localcontext() as arithmetic:
        arithmetic.prec = 80
        step_S010(ctx)
        ctx.finish()

File: mainframe_migration/examples/generated_targets/dotnet-bigquery/jobs/JOB002.cs
/* JOB002 — behavior-preserving migrated job.
 * Source: jcl/JOB002.jcl
 * Purpose: run every source step in order; preserve unusual rules.
 * Inputs and outputs: declared DD bindings in the sealed process model.
 * SQL operations use the selected database adapter; no source repair is applied.
 */
using System;
using System.Collections.Generic;

public static class JOB002 {
    // Execute S010 (SORT) using the source DD bindings.
    static void step_S010(MfRuntime.Context ctx) {
        var dd = ctx.dd("JOB002", "S010");
        ctx.sort(MfRuntime.obj(dd["SORTIN"]), MfRuntime.obj(dd["SORTOUT"]), new int[][] {new int[] {1,6,0,0}});
        ctx.end_step();
    }
    // Run all source steps exactly once. A failure stops rather than skips work.
    public static void run(MfRuntime.Context ctx) {
        step_S010(ctx);
        ctx.finish();
    }
}
